src/content/docs/setup/configuration/cors-setup.mdx

```diff
@@ -0,0 +1,124 @@
+---
+title: CORS Setup
+description: Quick guide to configuring CORS headers for Adobe Commerce GraphQL endpoints.
+---
+
+import { Aside } from '@astrojs/starlight/components';
+import Tasks from '@components/Tasks.astro';
+import Task from '@components/Task.astro';
+import Link from '@components/Link.astro';
+
+Cross-Origin Resource Sharing (CORS) is a browser security feature that restricts web pages from making requests to a different domain than the one serving the page. CORS is required when your storefront runs on a different domain or port than your Adobe Commerce backend. 
+
+<Aside type="caution" title="Production recommendation">
+CORS should be a last resort for production. Prefer same-origin delivery of the storefront and API (for example, via Fastly VCL) to avoid CORS entirely.
+
+But during development, it's a common scenario for storefronts to run on a different domain or port than the Adobe Commerce backend, so CORS is required. Without CORS, browsers will block cross-origin requests and you'll see errors like the following in the browser console:
+
+```txt
+Access to fetch at 'https://commerce.example.com/graphql' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present
+```
+</Aside>
+
+During development, it's common for storefronts to run on a different domain or port than the Adobe Commerce backend. For example, if your storefront is running on `http://localhost:3000` and your Adobe Commerce backend is running on `https://commerce.example.com`, you will need to configure CORS to allow requests from the storefront to the backend.
+
+## Quick test: Is CORS working?
+
+Your Adobe Commerce GraphQL endpoint (typically `/graphql`) is the primary endpoint that requires CORS configuration. If you already have CORS configured, test it by replacing `YOUR_STOREFRONT_URL` with your actual frontend URL:
+
+```bash
+curl -I -X OPTIONS https://your-commerce-backend.com/graphql \
+  -H "Origin: YOUR_STOREFRONT_URL" \
+  -H "Access-Control-Request-Method: POST"
+```
+
+Look for these headers in the response:
+- `Access-Control-Allow-Origin: YOUR_STOREFRONT_URL` (or `*` if using a wildcard)
+- `Access-Control-Allow-Methods: GET, POST, OPTIONS`
+
+If you see these headers with your storefront URL, CORS is configured correctly. If not, follow the setup steps below.
+
+## Quick steps
+
+<Tasks>
+
+<Task>
+
+### Add CORS support
+
+To add CORS headers to your Adobe Commerce GraphQL endpoints, you have several options:
+
+- **Server configuration**: Configure CORS headers at the web server level (Nginx, Apache)
+- **Custom module**: Implement a custom Adobe Commerce module to add CORS headers
+- **Third-party module**: Use a community module such as [graycore/magento2-cors](https://github.com/graycoreio/magento2-cors) or similar
+
+Each approach has different implementation steps. If using a module-based approach:
```

Comment:
I would not add installation steps for a 3rd party module.

Consider adding links for Nginx/Apache configuration. I found these

https://enable-cors.org/server_apache.html
https://enable-cors.org/server_nginx.html

Also, I would list custom module last. It's a lot harder than the other two options